asdf/_node_info.py

```diff
@@ -5,16 +5,41 @@
 from .treeutil import get_children
 
 
-def collect_schema_info(key, node, identifier="root", preserve_list=True, refresh_extension_manager=False):
+def collect_schema_info(key, path, node, identifier="root", preserve_list=True, refresh_extension_manager=False):
     """
-    Collect from the underlying schemas any of the info stored under key.
+    Collect from the underlying schemas any of the info stored under key, relative to the path
+
+    Parameters
+    ----------
+    key : str
+        The key to look up.
+    path : str or None
```

Comment:
I wonder about how lists are handled in all this with regard to path. Something I've entertained is using square brackets for lists in the path (dots not really needed). So turning a list into a dict means turning an integer key into a string I presume and implicit conversion back to an integer for indexing the list. But I don't see schema_info in master so do I need to review a different PR first?

Reply:
After looking at the corresponding code, it doesn't preclude such an enhancement in the future.